File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        int answer = 1;
        HashMap<String, Integer> map = new HashMap<>();
        for (String[] str : clothes) {
            map.put(str[1], map.getOrDefault(str[1], 0) + 1);
        }
        
        for (int value : map.values()) {
            answer = answer * (value + 1);
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    d = {}
    
    for i in clothes :
        if i[1] in d :
            d[i[1]].append(i[0])
        else :
            d[i[1]] = [i[0]]
    
    for i in d :
        answer *= len(d[i]) + 1

    return answer - 1